node/network/statement-distribution/src/lib.rs

```diff
@@ -620,13 +623,12 @@ async fn circulate_statement(
 ) -> Vec<PeerId> {
 	let fingerprint = stored.fingerprint();
 
-	let mut peers_to_send = HashMap::new();
+	let len_sqrt = (peers.len() as f64).sqrt() as usize;
+	let cap = std::cmp::max(MIN_GOSSIP_PEERS, len_sqrt);
 
-	for (peer, data) in peers.iter_mut() {
-		if let Some(new_known) = data.send(&relay_parent, &fingerprint) {
-			peers_to_send.insert(peer.clone(), new_known);
-		}
-	}
+	let peers_to_send: HashMap<PeerId, bool> = peers.iter_mut().filter_map(|(peer, data)| {
```

Comment:
There's no `choose_random_subset` here. Should there be?

Reply:
but that won't be a simplification, since we want to send only to peers for which send returns `Some`, and we don't want to not send after `data.send` return `Some`.

Reply:
HashMap random ordering is per run of the program, so it's not very high entropy. it'd be better to shuffle again.

Reply:
note that we take the first `n` to which we can send, not just the first `n`
also the iteration order changes when we insert and delete from the hashmap, so it's pretty much (pseudo)random